compute/instances/spot/create_spot.go

```diff
@@ -0,0 +1,153 @@
+	return &computepb.AttachedDisk{
+		AutoDelete: proto.Bool(autoDelete),
+		Boot:       proto.Bool(boot),
+		InitializeParams: &computepb.AttachedDiskInitializeParams{
+			DiskSizeGb:  proto.Int64(diskSizeGb),
+			DiskType:    proto.String(diskType),
+			SourceImage: proto.String(sourceImage),
+		},
+		Type: proto.String(computepb.AttachedDisk_PERSISTENT.String()),
+	}
+}
+
+// createInstance sends an instance creation request to the Compute Engine API and waits for it to complete.
+func createInstance(ctx context.Context, projectID, zone, instanceName string, disks []*computepb.AttachedDisk, machineType, networkLink string, spot bool) (*computepb.Instance, error) {
+	client, err := compute.NewInstancesRESTClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("NewInstancesRESTClient: %w", err)
+	}
+	defer client.Close()
+
+	networkInterface := &computepb.NetworkInterface{
+		Name: proto.String(networkLink),
+	}
+
+	instance := &computepb.Instance{
+		Name:              proto.String(instanceName),
+		Disks:             disks,
+		MachineType:       proto.String(fmt.Sprintf("zones/%s/machineTypes/%s", zone, machineType)),
+		NetworkInterfaces: []*computepb.NetworkInterface{networkInterface},
+		Scheduling: &computepb.Scheduling{
+			ProvisioningModel: proto.String(computepb.Scheduling_SPOT.String()),
+		},
+	}
+
+	req := &computepb.InsertInstanceRequest{
+		Project:          projectID,
+		Zone:             zone,
+		InstanceResource: instance,
+	}
+
+	op, err := client.Insert(ctx, req)
+	if err != nil {
+		return nil, fmt.Errorf("Insert: %w", err)
+	}
+
+	opClient, err := compute.NewZoneOperationsRESTClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("NewZoneOperationsRESTClient: %w", err)
+	}
+	defer opClient.Close()
+
+	if err = op.Wait(ctx); err != nil {
+		return nil, fmt.Errorf("unable to wait for the operation: %w", err)
+	}
+
+	return client.Get(ctx, &computepb.GetInstanceRequest{
+		Project:  projectID,
+		Zone:     zone,
+		Instance: instanceName,
+	})
+}
+
+// createSpotInstance creates a new Spot VM instance with Debian 10 operating system.
+func createSpotInstance(w io.Writer, projectID, zone, instanceName string) error {
+	// projectID := "your_project_id"
+	// zone := "europe-central2-b"
+	// instanceName := "your_instance_name"
+
+	ctx := context.Background()
+	image, err := getImageFromFamily("debian-cloud", "debian-11")
+	if err != nil {
+		return fmt.Errorf("getImageFromFamily: %w", err)
+	}
+
+	diskType := fmt.Sprintf("zones/%s/diskTypes/pd-standard", zone)
+	disks := []*computepb.AttachedDisk{
+		diskFromImage(diskType, 10, true, image.GetSelfLink(), true),
+	}
+
+	instance, err := createInstance(ctx, projectID, zone, instanceName, disks, "n1-standard-1", "global/networks/default", true)
+	if err != nil {
+		return fmt.Errorf("createInstance: %w", err)
+	}
+
+	fmt.Fprintf(w, "Instance created: %v\n", instance)
+	return nil
+}
+
+func main() {
```

Comment:
snippets should not have a main function. please remove this.

Reply:
fixed and moved to [PR](https://github.com/GoogleCloudPlatform/golang-samples/pull/4187)